packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright 2018-2021 BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers'
+import type {
+  Deposit,
+  IDidIdentifier,
+  IIdentity,
+  KeyringPair,
+  KeystoreSigner,
+  SubmittableExtrinsic,
+  VerificationKeyType,
+} from '@kiltprotocol/types'
+
+import type { SignerOptions } from '@polkadot/api/types'
+import type { Null, Option, Struct } from '@polkadot/types'
+import type { AccountId, Extrinsic } from '@polkadot/types/interfaces'
+import type { AnyNumber } from '@polkadot/types/types'
+import type { HexString } from '@polkadot/util/types'
+
+import type { DidAuthorizationOptions, FullDidDetails } from '../DidDetails'
+import { getKiltDidFromIdentifier, parseDidUri } from '../Did.utils.js'
+
+// TODO: update with string pattern types once available
+type AccountAddress = IIdentity['address']
+type DidUri = FullDidDetails['did']
+
+interface ConnectionRecord extends Struct {
+  did: AccountId
+  deposit: Deposit
+}
+
+interface DepositJSON {
```

Comment:
We have this defined in `types/src/Deposit.ts`.

Reply:
this was a decoded type but I guess we don't necessarily need to decode